test/error_messages.coffee

```diff
@@ -1317,3 +1317,24 @@ test "can't use pattern matches for loop indices", ->
     a for b, {c} in d
              ^^^
   '''
+
+test "bare 'super' is no longer allowed", ->
+  assertErrorFormat 'class extends A then constructor: -> super', '''
+    [stdin]:1:35: error: unexpected ->
+    class extends A then constructor: -> super
+                                      ^^
+  '''
+
+test "soaked 'super' in constructor", ->
+  assertErrorFormat 'class extends A then constructor: -> super?()', '''
+    [stdin]:1:38: error: Unsupported reference to 'super'
```

Comment:
Hrm, I copied the current node.js error message verbatim. I suspect that might be because if *is* a bit of a TODO in JS (given past discussions on ES about allowing bare `super`, amongst other things).

Since we're explicitly banning it, happy to change to "illegal" 👍